hermes/bin/src/runtime_extensions/hermes/crypto/bip39.rs

```diff
@@ -0,0 +1,391 @@
+    let mut word_index_vec = Vec::new();
+
+    // Separate entropy bits into 11 bits and convert to decimal.
+    // This decimal will be used to get the word index.
+    for i in 0..word_count {
+        let mut idx = 0;
+        for j in 0..11 {
+            if let Some(value) = bits_entropy.get(i * 11 + j) {
+                if *value > 0 {
+                    idx += 1 << (10 - j);
+                }
+            }
+        }
+        word_index_vec.push(idx);
+    }
+    word_index_vec
+}
+
+/// Get the mnemonic from the BIP39 dictionary using word indices.
+fn get_mnemonic_from_indices(word_index_vec: Vec<u16>, language: Language) -> Vec<String> {
+    let word_list = language.word_list();
+    let mut mnemonic: Vec<String> = vec![];
+    for word in word_index_vec {
+        if let Some(word) = word_list.get(word as usize) {
+            mnemonic.push((*(word)).to_string());
+        }
+    }
+    mnemonic
+}
+
+/// Turns decimal into binary array of length 11.
+fn decimal_to_binary_array(decimal: u16) -> [u8; 11] {
+    let mut binary = [0u8; 11];
+    let mut n = decimal;
+    let mut index = 0;
+
+    while n > 0 {
+        if let Some(value) = binary.get_mut(index) {
+            let bit = n % 2;
+            index += 1;
+            *value = bit as u8;
+            n /= 2;
+        }
+    }
+
+    // If the number of bits is less than 11, fill the remaining bits with 0.
+    while index < 11 {
+        if let Some(value) = binary.get_mut(index) {
+            *value = 0;
+        }
+        index += 1;
+    }
+    binary.reverse();
+    binary
+}
+
+/// Turns bit of array to bytes of arrays
+fn bits_to_bytes(bits: &[u8]) -> Vec<u8> {
+    bits.chunks(8)
+        .map(|chunk| {
+            chunk
+                .iter()
+                .enumerate()
+                .fold(0, |acc, (i, &bit)| acc | ((bit) << (7 - i)))
+        })
+        .collect()
+}
+
+/// Convert string to BIP39 language.
+fn string_to_language(s: &str) -> Language {
+    match s {
+        "SimplifiedChinese" => Language::SimplifiedChinese,
+        "TraditionalChinese" => Language::TraditionalChinese,
+        "Czech" => Language::Czech,
+        "French" => Language::French,
+        "Italian" => Language::Italian,
+        "Japanese" => Language::Japanese,
+        "Korean" => Language::Korean,
+        "Spanish" => Language::Spanish,
+        _ => Language::English,
+    }
+}
+
+#[cfg(test)]
+mod tests_bip39 {
+    use super::*;
+
+    // English test vector is from https://cips.cardano.org/cip/CIP-0011
+    // Entropy can be checked from https://iancoleman.io/bip39
+    const MNEMONIC_ENG: &str = "prevent company field green slot measure chief hero apple task eagle sunset endorse dress seed";
+    // Japanese test is from https://github.com/rust-bitcoin/rust-bip39/blob/master/src/lib.rs
+    const MNEMONIC_JAPANESE: &str = "こころ いどう きあつ そうがんきょう へいあん せつりつ ごうせい はいち  いびき きこく あんい おちつく きこえる けんとう たいこ すすめる はっけん ていど はんおん いんさつ うなぎ しねま れいぼう みつかる";
```

Comment:
I think you can make this separate files and include them.

Reply:
Since it is just a test vector, I think it is better to add it here